torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -495,14 +495,26 @@ struct NCCLTraceBuffer {
     return result;
   }
 
-  void retire_id(c10::optional<size_t> id) {
+  // Don't set compute_duration=True from inside dump API,
```

Comment:
I think `retire_id` is only called in one place, and compute_duration will always be true. But dump_entries also calls getDurationFromFirstEvent, and that isn't prevented by this flag. So if we end up dumping an event that just became, done but hasn't been retired, then we still see the dumper thread try to calculate a duration. I was thinking that this change would add a flag to `update_state` so that when called from `dump_entries` it would not calculate the duration, but if called from retire_id it would.

Reply:
oh - yes i intended to delete the call to `getDurationFromFirstEvent` from inside `update_state`, and have only retire_id calling duration.  Thanks for catching that.